Packs/PaloAltoNetworks_IoT3rdParty/Integrations/PaloAltoNetworksIoT3rdParty/PaloAltoNetworksIoT3rdParty.py

```diff
@@ -0,0 +1,758 @@
+                data = convert_vulnerability_to_servicenow(demisto.args())
+                readable_res = "Converted an Vulnerability has zb_ticketid %s to %s" % (
+                    demisto.args().get('metadata1')['correlation_id'], output_format)
+        else:
+            return_error("Output format ServiceNow not supported for %s" % asset_type)
+
+    elif output_format == "SIEM":
+        if asset_list:
+            if asset_type == "Device":
+                data = convert_device_list_to_cef(asset_list)
+            elif asset_type == 'Alert':
+                data = convert_alert_list_to_cef(asset_list)
+            elif asset_type == "Vulnerability":
+                data = convert_vulnerability_list_to_cef(asset_list)
+            else:
+                return_error("Output format SIEM not supported for %s" % asset_type)
+            readable_res = "Converted %d %s to %s" % (len(data), asset_type, output_format)
+
+    elif output_format == 'CiscoISECustomAttributes':
+        if asset_list:
+            if asset_type == 'Device':
+                data = convert_device_list_to_ise_attributes(asset_list)
+        else:
+            return_error("Output format CiscoISECustomAttributes not supported for %s" % asset_type)
+        readable_res = "Converted %d %s to %s" % (len(data), asset_type, output_format)
+
+    prefix = prefixMap[asset_type][output_format]
+
+    return CommandResults(
+        readable_output=readable_res,
+        outputs_prefix=prefix,
+        outputs=data
+    )
+
+
+def connection_test_command(return_bool: bool = False):
+    """
+    Try to get a single device from the Cloud to test connectivity.
+    """
+    params = {
+        'offset': '0',
+        'pagelength': '1',
+        'detail': 'false',
+    }
+    url = BASE_URL + api_type_map['Device']['list_url']
+    http_request('GET', url, params)
+
+    if return_bool:
+        return True
+
+
+def main() -> None:
+    """main function, parses params and runs command functions
+    :return:
+    :rtype:
+    """
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        if demisto.command() == 'test-module':
+            if connection_test_command(True):
+                demisto.results('ok')
+            else:
+                demisto.results('test failed')
+        elif demisto.command() == 'panw-iot-3rd-party-get-asset-list':
+            results = get_asset_list()
+            return_results(results)
+        elif demisto.command() == 'panw-iot-3rd-party-get-single-asset':
+            results = get_single_asset()
+            return_results(results)
+        elif demisto.command() == 'panw-iot-3rd-party-report-status-to-panw':
+            results = report_status_to_iot_cloud()
+            return_results(results)
+        elif demisto.command() == 'panw-iot-3rd-party-convert-assets-to-external-format':
+            results = convert_asset_to_external_format()
+            return_results(results)
```

Comment:
please move the demisto.args() to the funcs and specify the args name in the cmd signature.